WordPress/src/main/java/org/wordpress/android/ui/domainregister/suggestionslist/DomainSuggestionsAdapter.kt

```diff
@@ -0,0 +1,36 @@
+package org.wordpress.android.ui.domainregister.suggestionslist
+
+import android.support.v7.widget.RecyclerView.Adapter
+import android.view.ViewGroup
+import org.wordpress.android.fluxc.network.rest.wpcom.site.DomainSuggestionResponse
+
+class DomainSuggestionsAdapter(
+    private val itemSelectionListener: (DomainSuggestionResponse?, Int) -> Unit
+) : Adapter<DomainSuggestionsViewHolder>() {
+    private val list = mutableListOf<DomainSuggestionResponse>()
+    var selectedPosition = -1
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DomainSuggestionsViewHolder {
+        return DomainSuggestionsViewHolder(parent, this::onDomainSuggestionSelected)
+    }
+
+    override fun getItemCount(): Int {
+        return list.size
+    }
+
+    override fun onBindViewHolder(holder: DomainSuggestionsViewHolder, position: Int) {
+        holder.bind(list[position], position, selectedPosition == position)
+    }
+
+    private fun onDomainSuggestionSelected(suggestion: DomainSuggestionResponse?, position: Int) {
+        selectedPosition = position
+        notifyDataSetChanged()
+        itemSelectionListener(suggestion, position)
+    }
+
+    internal fun updateSuggestionsList(items: List<DomainSuggestionResponse>) {
+        list.clear()
+        list.addAll(items)
+        notifyDataSetChanged()
```

Comment:
Since we are resetting the whole list, it might be easier and cheaper (performance wise) to call `notifyDataSetChanged()`. 
`notifyItemRangeChanged` will work as well, but the size of the new suggestions list might be different, which might cause problems. What do you think?

Reply:
Fair enough;)!